source/common/grpc/json_transcoder_filter.cc

```diff
@@ -0,0 +1,397 @@
+
+#include "google/api/annotations.pb.h"
+#include "google/api/http.pb.h"
+#include "google/protobuf/descriptor.h"
+#include "google/protobuf/descriptor.pb.h"
+#include "google/protobuf/util/type_resolver.h"
+#include "google/protobuf/util/type_resolver_util.h"
+#include "grpc_transcoding/json_request_translator.h"
+#include "grpc_transcoding/response_to_json_translator.h"
+
+using google::grpc::transcoding::JsonRequestTranslator;
+using google::grpc::transcoding::RequestInfo;
+using google::grpc::transcoding::ResponseToJsonTranslator;
+using google::grpc::transcoding::Transcoder;
+using google::grpc::transcoding::TranscoderInputStream;
+using google::protobuf::DescriptorPool;
+using google::protobuf::FileDescriptor;
+using google::protobuf::FileDescriptorSet;
+using google::protobuf::io::ZeroCopyInputStream;
+using google::protobuf::util::error::Code;
+using google::protobuf::util::Status;
+
+namespace Envoy {
+namespace Grpc {
+
+namespace {
+
+const std::string TypeUrlPrefix{"type.googleapis.com"};
+
+// Transcoder:
+// https://github.com/grpc-ecosystem/grpc-httpjson-transcoding/blob/master/src/include/grpc_transcoding/transcoder.h
+// implementation based on JsonRequestTranslator & ResponseToJsonTranslator
+class TranscoderImpl : public Transcoder {
+public:
+  /**
+   * Construct a transcoder implementation
+   * @param request_translator a JsonRequestTranslator that does the request translation
+   * @param response_translator a ResponseToJsonTranslator that does the response translation
+   */
+  TranscoderImpl(std::unique_ptr<JsonRequestTranslator> request_translator,
+                 std::unique_ptr<ResponseToJsonTranslator> response_translator)
+      : request_translator_(std::move(request_translator)),
+        response_translator_(std::move(response_translator)),
+        request_stream_(request_translator_->Output().CreateInputStream()),
+        response_stream_(response_translator_->CreateInputStream()) {}
+
+  // Transcoder
+  ::google::grpc::transcoding::TranscoderInputStream* RequestOutput() {
+    return request_stream_.get();
+  }
+  Status RequestStatus() { return request_translator_->Output().Status(); }
+
+  ZeroCopyInputStream* ResponseOutput() { return response_stream_.get(); }
+  Status ResponseStatus() { return response_translator_->Status(); }
+
+private:
+  std::unique_ptr<JsonRequestTranslator> request_translator_;
+  std::unique_ptr<ResponseToJsonTranslator> response_translator_;
+  std::unique_ptr<TranscoderInputStream> request_stream_;
+  std::unique_ptr<TranscoderInputStream> response_stream_;
+};
+
+} // namespace
+
+JsonTranscoderConfig::JsonTranscoderConfig(const Json::Object& config) {
+  const std::string proto_descriptor_file = config.getString("proto_descriptor");
+  FileDescriptorSet descriptor_set;
+  if (!descriptor_set.ParseFromString(Filesystem::fileReadToEnd(proto_descriptor_file))) {
+    throw EnvoyException("transcoding_filter: Unable to parse proto descriptor");
```

Comment:
I don't see any `EXPECT_THROW` or `EXPECT_THROW_WITH_MESSAGE` statements below. Are we testing and obtaining coverage of this error handling (and below)?